receiver/splunkenterprisereceiver/scraper.go

```diff
@@ -52,36 +53,86 @@ func (s *splunkScraper) start(_ context.Context, h component.Host) (err error) {
 	return nil
 }
 
+// listens to the error channel and combines errors sent from different metric scrape functions,
+// returning the combined error list should context timeout or a nil error value is sent in the
+// channel signifying the end of a scrape cycle
+func errorListener(ctx context.Context, eQueue <-chan error, eOut chan<- *scrapererror.ScrapeErrors) {
+	errs := &scrapererror.ScrapeErrors{}
+
+	for {
+		select {
+		// context timeout
+		case <-ctx.Done():
+			eOut <- errs
+			return
+		case err, ok := <-eQueue:
+			// shutdown
+			if err == nil || !ok {
+				eOut <- errs
+				return
+				// or add an error to errs
+			} else {
+				errs.Add(err)
```

Comment:
is that threadsafe?

Reply:
```
// Add adds a regular error.
func (s *ScrapeErrors) Add(err error) {
	s.errs = append(s.errs, err)
}
```

It is not threadsafe.

Reply:
How about accumulating errors in the channel and emptying the channel all at once when the scrape is done?

Reply:
sorry if im being a little dense but do i need to be worried about the threadsafety of `Add()` in this case? there is only one function which accesses the `errs` object and it does so in a single routine.